src/maps.rs

```diff
@@ -108,17 +162,26 @@ fn parse_maps_line<'line>(line: &'line str, pid: Pid) -> Result<MapsEntry> {
     let path_str = split_once(line, "inode component")
         .map(|(_inode, line)| line.trim())
         .unwrap_or("");
-    let path = if path_str.ends_with(" (deleted)") {
-        PathBuf::from(format!("/proc/{pid}/map_files/{address_str}"))
-    } else {
-        PathBuf::from(path_str)
+
+    let path_name = match path_str.as_bytes() {
+        [] => None,
+        [b'[', .., b']'] => Some(PathName::Component(path_str.to_string())),
```

Comment:
it's not always `[something]` for non-file entries, so best not to assume this format. I think just checking `/` at the beginning should be fine to distinguish files on disk from anything special. And later on we can explicitly check for `[vdso]` (as an example), when we need to process them specially.

Here are few examples from real process on my machine:

```
7ff8d9eab000-7ff8d9ecc000 rw-s 00000000 00:0e 2057                       anon_inode:[perf_event]
7ff8d9ecc000-7ff8d9eed000 rw-s 00000000 00:0e 2057                       anon_inode:[perf_event]
7ff8d9f2d000-7ff8d9f2e000 r--s 00000000 00:0e 2057                       anon_inode:bpf-map
7ff8d9f6f000-7ff8d9f70000 r--s 00000000 00:0e 2057                       anon_inode:bpf-map
```

Reply:
I adjusted the code accordingly. Thanks!